composition-js/src/composeDirectiveManager.ts

```diff
@@ -174,6 +175,7 @@ export class ComposeDirectiveManager {
       this.pushHint(new CompositionHint(
         HINTS.DIRECTIVE_COMPOSITION_INFO,
         `Directive "@${directive.name}" should not be explicitly manually composed since it is a federation directive composed by default`,
+        `@${directive.name}`,
```

Comment:
I think we should avoid manually crafting any coordinate when the [`coordinate`](https://github.com/apollographql/federation/blob/main/internals-js/src/definitions.ts#L671) property exists.

In fact, see my suggestion on the ctor of `CompositionHint` which would pretty much guarantee this by design.